job-server/src/main/scala/spark/jobserver/io/FileCacher.scala

```diff
@@ -27,21 +27,45 @@ trait FileCacher {
     appName + "-" + uploadTime.toString("yyyyMMdd_hhmmss_SSS") + s".${binaryType.extension}"
   }
 
+  // Random temporary filename suffix possible chars
```

Comment:
I don't see your reasoning for wanting to control the random part, it's a temp file, it really shouldn't matter, but you can add whatever you want to the suffix and prefix. Also the second point, you can control the file location, it's there in the documentation.